src/helm/proxy/clients/client.py

```diff
@@ -118,3 +118,19 @@ def truncate_sequence(sequence: Sequence, request: Request, print_warning: bool
         sequence = Sequence(text=new_text, logprob=new_logprob, tokens=new_tokens)
 
     return sequence
+
+
+def cleanup_str(token: str, tokenizer_name: Optional[str] = None) -> str:
+    """
+    Certain tokenizers introduce special characters to represent spaces, such as
+    "Ġ" or "▁". This function removes those characters.
+    """
+    # TODO: tokenizer_name is here in case some models need special handling.
+    return token.replace("Ġ", " ").replace("▁", " ")
```

Comment:
I think you want to only do `Ġ`, or only do `_`, depending on the tokenizer name. Otherwise you end up with unintended modifications. I'd suggest just putting a list of tokenizer names (or patterns) here that need modifications, and figure out which replacement to do.

Or, you could have the a caller pass in an enum (byte pair / sentencepiece) as an argument to tell it which replacement to do, if the caller should know.

Eventually we probably want some concept of tokenizer tag.

Reply:
I tried to separate models. This look hackier IMO, but I think this is correct